rust/hyperlane-base/src/contract_sync/cursor.rs

```diff
@@ -30,43 +36,40 @@ where
             chunk_size,
             last_tip_update: Instant::now(),
             from: initial_height,
+            eta_calculator: SyncerEtaCalculator::new(initial_height, tip, ETA_TIME_WINDOW),
         })
     }
 
     /// Wait based on how close we are to the tip and update the tip,
     /// i.e. the highest block we may scrape.
     async fn rate_limit(&mut self) -> ChainResult<()> {
+        let update_tip = self.last_tip_update.elapsed() >= Duration::from_secs(30);
         if self.from + self.chunk_size < self.tip {
             // If doing the full chunk wouldn't exceed the already known tip,
             // we don't necessarily need to fetch the new tip. Sleep a tiny bit
             // so that we can catch up to the tip relatively quickly.
-            sleep(Duration::from_secs(1)).await;
-            Ok(())
-        } else {
+            sleep(Duration::from_millis(100)).await;
+        } else if !update_tip {
             // We are close to the tip.
-            if self.last_tip_update.elapsed() < Duration::from_secs(30) {
-                // Sleep a little longer because we have caught up.
-                sleep(Duration::from_secs(10)).await;
-            } else {
-                // We are probably not caught up yet. This would happen if we
-                // started really far behind so now it is very likely the tip
-                // has moved a significant distance. We don't want to wait in
-                // this case any more than we normally would.
-                sleep(Duration::from_secs(1)).await;
-            }
+            // Sleep a little longer because we have caught up.
+            sleep(Duration::from_secs(10)).await;
+        }
 
-            match self.indexer.get_finalized_block_number().await {
-                Ok(tip) => {
-                    // we retrieved a new tip value, go ahead and update.
-                    self.last_tip_update = Instant::now();
-                    self.tip = tip;
-                    Ok(())
-                }
-                Err(e) => {
-                    // we are failing to make a basic query, we should wait before retrying.
-                    sleep(Duration::from_secs(10)).await;
-                    Err(e)
-                }
+        if !update_tip {
+            return Ok(());
+        }
+        match self.indexer.get_finalized_block_number().await {
+            Ok(tip) => {
+                // we retrieved a new tip value, go ahead and update.
+                self.last_tip_update = Instant::now();
+                self.tip = tip;
+                Ok(())
+            }
+            Err(e) => {
+                warn!(error = %e, "Failed to get next block range");
```

Comment:
don't feel strongly but wondering if we should also mention that it's because we were unable to get the finalized block number?